libs/myReact/useEffect.ts

```diff
@@ -0,0 +1,53 @@
+type Callback = () => void | (() => void)
+
+const effects: { deps?: any[] }[] = []
+const effectsArrayAfterRender: { callback: Callback; isRuned?: boolean }[] = []
+let effectsIdx = 0
+
+function isDifferentDeps(prevDeps: any[], currDeps: any[]): boolean {
+	return prevDeps.some((item, i) => !Object.is(item, currDeps[i]))
+}
+
+export default function useEffect(
+	callback: () => void | (() => void),
```

Comment:
여기도 만들어주신 타입을 사용해주면 좋겠네요~
```suggestion
	callback: Callback,
```